File: Domain/RuleProject.cs
namespace Domain
{
    public class RuleProject
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<RuleProperty> Properties { get; set; }
        public ICollection<Rule> StandardRules { get; set; }
        public ICollection<DecisionTable> DecisionTables { get; set; }
        public ICollection<RuleProjectMember> Members { get; set; } = new List<RuleProjectMember>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: client-app/src/app/models/ruleProject.ts
import { DecisionTable } from "./decisionTable";
import { Profile } from "./profile";
import { Rule } from "./rule";

export interface IRuleProject {
    id: string;
    name: string;
    description: string;
    createdAt: Date;
    properties: RuleProperty[];
    standardRules: Rule[];
    decisionTables: DecisionTable[];
    members: Profile[];
    owner: string;
}

export interface RuleProperty {
    id: string;
    name: string;
    type: string;
    direction: string;
    subProperties: RuleProperty[];
}

export class RuleProject implements IRuleProject {
    constructor(init: RPFormValues) {
        this.id = init.id!
        this.name = init.name
        this.description = init.description
    }

    id: string;
    name: string;
    description: string;
    createdAt: Date = new Date();
    properties: RuleProperty[] = [];
    standardRules: Rule[] = [];
    decisionTables: DecisionTable[] = [];
    members: Profile[] = [];
    owner: string = '';
}

export class RPFormValues {
    id?: string = undefined;
    name: string = '';
    description: string = '';

    constructor(table?: RPFormValues) {
        if (table) {
            this.id = table.id;
            this.name = table.name;
            this.description = table.description;
        }
    }
}